new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Pod.java

```diff
@@ -0,0 +1,200 @@
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+public class Pod {
+
+  // reusable condition factory
+  private static ConditionFactory retry
+      = with().pollInterval(5, SECONDS).atMost(10, MINUTES).await();
+
+  /**
+   * Check the pods in the given namespace are restarted in a rolling fashion.
+   * Waits until all pods are restarted, for upto 10 minutes.
+   *
+   * @param domainUid UID of the WebLogic domain
+   * @param namespace name of the namespace in which to check for the pods restart sequence
+   * @return true if pods in the namespace are restarted in a rolling fashion otherwise false
+   * @throws ApiException when Kubernetes cluster query fails
+   * @throws InterruptedException when pod status check threads are interrupted
+   * @throws ExecutionException when pod status checks times out
+   * @throws TimeoutException when waiting for the threads times out
+   */
+  public static boolean isARollingRestart(String domainUid, String namespace)
+      throws ApiException, InterruptedException, ExecutionException, TimeoutException {
+
+    // query cluster and get pods from the namespace
+    String labelSelectors = "weblogic.serverName";
+    V1PodList listPods = Kubernetes.listPods(namespace, labelSelectors);
+    ArrayList<String> podNames = new ArrayList<>();
+
+    //return if no pods are found
+    if (listPods.getItems().isEmpty()) {
+      logger.severe("No pods found in namespace {0}", namespace);
+      return false;
+    } else {
+      logger.info("WebLogic pods found in namespace {0}", namespace);
+      for (V1Pod item : listPods.getItems()) {
+        logger.info(item.getMetadata().getName());
+        podNames.add(item.getMetadata().getName());
+      }
+    }
+
+    // check the pods termination status in concurrent thread
+    ExecutorService executorService = Executors.newFixedThreadPool(podNames.size());
+    ArrayList<Future<Boolean>> threads = new ArrayList<Future<Boolean>>();
+    for (var podName : podNames) {
+      // check for pod termination status and return true if pod is terminating
+      threads.add(executorService.submit(() -> {
+        retry
+            .conditionEvaluationListener(
+                condition -> logger.info("Waiting for pod {0} in namespace {1} to terminate"
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                    podName,
+                    namespace,
+                    condition.getElapsedTimeInMS(),
+                    condition.getRemainingTimeInMS()))
+            .until(onlyGivenPodTerminating(podName, domainUid, namespace));
+        return true;
+      }));
+      // wait for the callable to finish running and check if all pods were terminating
+      for (var future : threads) {
+        if (!future.get(10, MINUTES)) {
+          return false;
+        }
+      }
+    }
+    executorService.shutdownNow();
+
+    // wait for pods to become ready
+    for (var podName : podNames) {
```

Comment:
do you need to wait for them to be ready, or just running?

Reply:
The refactored code will check the termination status and then the ready status